source/game/field/obj/ObjectShip64.cc

```diff
@@ -0,0 +1,83 @@
+#include "ObjectShip64.hh"
+#include <egg/math/Matrix.hh>
+#include <egg/math/Vector.hh>
+#include <game/field/ObjectCollisionCylinder.hh>
+#include <game/field/RailInterpolator.hh>
+#include <game/field/obj/ObjectBase.hh>
+#include <game/field/obj/ObjectCollidable.hh>
+
+namespace Field {
+
+ObjectShip64::ObjectShip64(const System::MapdataGeoObj &params) : ObjectCollidable(params) {}
+
+void ObjectShip64::init() {
+    m_railInterpolator->init(0, 0);
+
+    m_tangent = m_railInterpolator->curTangentDir();
+    m_railInterpolator->setPerPointVelocities(true);
+
+    calc();
+}
+
+void ObjectShip64::calc() {
+    m_railInterpolator->calc();
+
+    setPos(m_railInterpolator->curPos());
+
+    fprintf(stderr, "m_pos: %f, %f, %f\n", m_pos.x, m_pos.y, m_pos.z);
+
+    EGG::Vector3f v;
+    v = m_tangent - m_railInterpolator->curTangentDir();
+    v *= 0.2;
+    v += m_tangent;
+    m_tangent = v;
+    m_tangent.normalise();
```

Comment:
This uses an interpolation function present in `ObjectBase`.
```suggestion
    m_tangent= Interpolate(0.2f, m_bc, m_railInterpolator->curTangentDir());
    m_tangent.normalise();
```

Reply:
ohh! that's what it is called